Lib/difflib.py

```diff
@@ -1054,6 +1052,8 @@ def IS_LINE_JUNK(line, pat=re.compile(r"\s*(?:#\s*)?$").match):
     False
     """
 
+    if pat is None:
+        return line.strip() in '#'
```

Comment:
`x in '#'` matches the empty string, but this isn't immediately obvious.
```suggestion
        return line.strip() in {'', '#'}
```